src/lib/datepicker/datepicker.spec.ts

```diff
@@ -25,50 +33,95 @@ import {MatDatepickerInput} from './datepicker-input';
 import {MatDatepickerToggle} from './datepicker-toggle';
 import {MAT_DATEPICKER_SCROLL_STRATEGY, MatDatepickerIntl, MatDatepickerModule} from './index';
 
+
+// NgModules used in `configureTestingModule` for each top-level `describe` block.
+const commonNgModuleImports = [
+  FormsModule,
+  MatDatepickerModule,
+  MatFormFieldModule,
+  MatInputModule,
+  NoopAnimationsModule,
+  ReactiveFormsModule,
+];
+
 describe('MatDatepicker', () => {
+  /** Whether the native Intl APIs are supported. */
   const SUPPORTS_INTL = typeof Intl != 'undefined';
 
-  // Creates a test component fixture.
-  function createComponent(
-    component: Type<any>,
-    imports: Type<any>[] = [],
-    providers: (FactoryProvider | ValueProvider)[] = [],
-    entryComponents: Type<any>[] = []): ComponentFixture<any> {
+  /**  Fake subject for signaling that a scroll has occurred. */
+  let scrolledSubject: Subject<void>;
+
+  /** Fake `Directionality` service. */
+  let fakeDirectectionality: Partial<Directionality>;
+
+  /** Fake `change` stream for the fake `Directionality` service */
+  let fakeDirectionChanges: EventEmitter<Direction>;
+
+  // Always clean up the overlay container after every test.
+  afterEach(inject([OverlayContainer], (container: OverlayContainer) => {
+    container.ngOnDestroy();
+  }));
+
+  beforeEach(async(() => {
+    scrolledSubject = new Subject();
+    fakeDirectionChanges = new EventEmitter<Direction>();
+    fakeDirectectionality = {
+      value: 'ltr',
+      change: fakeDirectionChanges,
+    };
 
     TestBed.configureTestingModule({
       imports: [
-        FormsModule,
-        MatDatepickerModule,
-        MatFormFieldModule,
-        MatInputModule,
-        NoopAnimationsModule,
-        ReactiveFormsModule,
-        ...imports
+        commonNgModuleImports,
+        MatNativeDateModule,
       ],
-      providers,
-      declarations: [component, ...entryComponents],
-    });
-
-    TestBed.overrideModule(BrowserDynamicTestingModule, {
+      declarations: [
+        StandardDatepicker,
```

Comment:
I think we decided that compiling all of the components was making our tests super slow, that's why we switched from something like this to what we currently have